src/components/presentational/TimeSeriesChart/TimeSeriesChart.tsx

```diff
@@ -1,57 +1,48 @@
 import React, { useContext } from 'react'
 import { add, addDays, addMonths, format, isToday } from 'date-fns'
 import { CardTitle, LayoutContext, LoadingIndicator } from '..'
-import { Area, AreaChart, Bar, BarChart, CartesianGrid, Cell, Line, LineChart, ReferenceLine, ResponsiveContainer, Tooltip, XAxis, YAxis } from 'recharts'
-import "./TimeSeriesChart.css"
+import { Area, Bar, CartesianGrid, Cell, ComposedChart, Line, ReferenceLine, ResponsiveContainer, Tooltip, XAxis, YAxis } from 'recharts'
+import './TimeSeriesChart.css'
 import { AxisDomain } from 'recharts/types/util/types'
-import { ColorDefinition, resolveColor } from '../../../helpers/colors'
+import { AreaChartSeries, ChartSeries, ColorDefinition, createAreaChartDefs, createBarChartDefs, createLineChartDefs, MultiSeriesBarChartOptions, MultiSeriesLineChartOptions, resolveColor } from '../../../helpers'
 import getDaysInMonth from 'date-fns/getDaysInMonth'
 import { ceil } from 'lodash'
 import addHours from 'date-fns/addHours'
 import startOfMonth from 'date-fns/startOfMonth'
-import { AreaChartSeries, ChartSeries, MultiSeriesBarChartOptions, MultiSeriesLineChartOptions } from '../../../helpers/chartOptions'
-import { createLineChartDefs } from '../../../helpers'
 
 export interface TimeSeriesDataPoint {
     timestamp: number; // Unix Timestamp in ms since epoch
-    // Other properties of this object are either either:
+    // Other properties of this object are either:
     //  - Named the same as a dataKey property of the configured ChartSeries for the graph
     //  - Arbitrary properties you want access to in places such as the tooltip callback.
     [key: string]: any;
 }
 
-
 export interface TimeSeriesChartProps {
-    title?: string
-    intervalType?: "Week" | "Month" | "6Month" | "Day",
-    intervalStart: Date,
-    data: TimeSeriesDataPoint[] | undefined,
-    expectedDataInterval?: Duration,
-    series: ChartSeries[] | AreaChartSeries[],
-    chartHasData: boolean,
-    tooltip: ({ active, payload, label }: any) => React.JSX.Element | null,
-    chartType: "Line" | "Bar" | "Area"
-    options?: MultiSeriesLineChartOptions | MultiSeriesBarChartOptions,
-    syncId?: string,
-    innerRef?: React.Ref<HTMLDivElement>
+    title?: string;
+    intervalType?: "Week" | "Month" | "6Month" | "Day";
+    intervalStart: Date;
+    data?: TimeSeriesDataPoint[];
+    expectedDataInterval?: Duration;
+    series: ChartSeries[] | AreaChartSeries[];
+    chartHasData: boolean;
+    tooltip?: ({ active, payload, label }: any) => React.JSX.Element | null;
+    chartType: "Line" | "Bar" | "Area";
+    options?: MultiSeriesLineChartOptions | MultiSeriesBarChartOptions;
+    syncId?: string;
+    children?: React.ReactNode;
```

Comment:
Nit: not thrilled with this variable name if it will only ever be overlay. I see it named the same in one of the recharts stories, though in that case children could be y/x axis, tooltip, series data.

Reply:
This, as far as I know, has to be named this way in order for React to properly pass you child DOM elements: 
 https://react.dev/reference/react/Children.  i.e. it's what allows this structure:

```
<TimeSeriesChart>
    <Bar>...</Bar>
    <Area>....</Area>
</TimeSeriesChart>
```

If you named it something else you'd be forced to do:
```
<TimeSeriesChart overlays={[<Bar>...</Bar>, <Area>...</Area>]}/>
```

@greinard I don't know why I didn't originally ask this when we were on a call last week, but - the whole reason you want this sort of functionality rather than, say, moving the `chartType` property onto `ChartSeries` property (which would allow you to specify that you want to show one series as a Line, one series as a Bar) is that you're basically exposing explicit control over the exact rendering of the overlayed charts, right?  

In some ways, while the second format is ugly, I can kind of see @reneefromhold's point that the mixed usage is slightly confusing.  It does, in some ways, open the door to passing *no* `series` to the chart, but then passing `children` and expecting something sane to happen (which, I think at this point it would not).  Giving it an explicitly named parameter *does* sort of nudge the usage in a particular direction and could be considered slightly more "self documenting."  

I'm not sure I feel super strongly one way or the other - as we sort of discussed in our chat, this wades into the territory of how much good are we doing by "wrapping" the charting library versus explicitly exposing it.

Reply:
Yeah.  As mentioned, the `children` property is just the built-in naming convention to be able to use the nested tags, and I think we want to retain that capability.

I also think that name works okay in that I happen to be using it for what I'm calling an "overlay", but really, it is just allowing a user of the `TimeSeriesChart` to add any supported recharts child component for rendering without always needing to update the `TimeSeriesChart` API to support a specific need.

Moving the chart type to the series may help cover the bars with lines use case that I have right now, but allowing generic children to be added to the `TimeSeriesChart` also allows for adding custom `ReferenceArea`, `ReferenceLine`, `Brush`, `Legend`, `Line`, `Area`, `Bar`, `Customized`, etc tags as needed.

We may want to consider adding first class support for those kinds of things to the `TimeSeriesChart`, but for the time being, having the `children` property will allow for custom chart implementations without the need to expand upon the shared component.  And yes, I do have some reservations about how far we continue down the path of wrapping rechart components rather than just using them directly.  🙂